src/platform/unix/mod.rs

```diff
@@ -947,11 +947,9 @@ fn recv(fd: c_int, blocking_mode: BlockingMode)
     Ok((main_data_buffer, channels, shared_memory_regions))
 }
 
-#[cfg(not(all(target_os="linux", feature="memfd")))]
+#[cfg(not(any(target_os="freebsd", all(target_os="linux", feature="memfd"))))]
 fn create_shmem(name: CString, length: usize) -> c_int {
     unsafe {
-        // NB: the FreeBSD man page for shm_unlink states that it requires
-        // write permissions, but testing shows that read-write is required.
         let fd = libc::shm_open(name.as_ptr(),
                                 libc::O_CREAT | libc::O_RDWR | libc::O_EXCL,
                                 0o600);
```

Comment:
You should be able to decrease this in the linux specific `shm_open`. Last time I tested this, FreeBSD required read-write, but it was not required on linux.

Reply:
Yeah, I thought about that. It's not just for Linux though, it's for everything else, and I'm not sure where write-only would be ok. Linux has `memfd` anyway…

Reply:
Ah, true

Reply:
`memfd` usage is optional on Linux, though. (IIRC that was decided because `memfd` is a pretty recent addition, so it would fail on older Linux versions...)

As for other systems, it's entirely unclear. Actually, I'm not sure we can even rely on it Linux: while it seems more logical to require only write permissions, neither the Linux nor the POSIX man pages seem to mention anything about this at all... I guess it's safer to stick with the broader permissions -- it's not like it has any real downsides here, aside from being confusing.

Either way, I don't think it's appropriate to remove the comment entirely. Even if the code is no longer used on FreeBSD, it doesn't make the comment wrong -- and we need to keep *some* explanation for why the code is this way...

Reply:
@antrik well I moved the comment to the `#[cfg(target_os="freebsd")]` section, I did not remove it entirely :)

Reply:
@myfreeweb I am aware that you moved it -- but since you didn't modify the original code, it's still not valid to just drop the comment entirely... Feel free to change the wording though, if you feel the original comment would be confusing in view of the changed situation.